bundler/lib/dependabot/bundler/file_updater.rb

```diff
@@ -51,11 +51,57 @@ def updated_dependency_files
         end
 
         check_updated_files(updated_files)
+
+        base_dir = updated_files.first.directory
+        updated_vendor_cache_files(base_directory: base_dir).each do |file|
+          updated_files << file
+        end
+
         updated_files
       end
 
       private
 
+      # Dynamically fetch the vendor cache folder from bundler
+      def vendor_cache_dir
+        return @vendor_cache_dir if defined?(@vendor_cache_dir)
+
+        @vendor_cache_dir =
+          SharedHelpers.in_a_forked_process do
+            # Set the path for path gemspec correctly
+            ::Bundler.instance_variable_set(:@root, repo_contents_path)
+            ::Bundler.app_cache
+          end
+      end
+
+      # Returns changed files in the vendor/cache folder
+      #
+      # @param base_directory [String] Update config base directory
+      # @return [Array<Dependabot::DependencyFile>]
+      def updated_vendor_cache_files(base_directory:)
+        return [] unless repo_contents_path && vendor_cache_dir
+
+        Dir.chdir(repo_contents_path) do
+          relative_dir = vendor_cache_dir.sub("#{repo_contents_path}/", "")
+          status = SharedHelpers.run_shell_command(
+            "git status --porcelain=v1 #{relative_dir}"
+          )
+          changed_paths = status.split("\n").map { |l| l.split(" ") }
+          changed_paths.map do |type, path|
+            deleted = type == "D"
+            encoding = Dependabot::DependencyFile::ContentEncoding::BASE64
```

Comment:
You're suggestion it would be `null` in this case, or we remove the key entirely?

I suppose having a `content_encoding` doesn't make a ton of sense if we don't have any `content` to encode.

Reply:
The 27 number was assuming the key was removed (e.g. `omitempty` in golang), 🤷  for how Ruby's JSON.

Ignoring byte counts: I'd dig `nil` or `""` as a placeholder when there is no `content`

Reply:
Maybe overthinking this, what about a `ContentEncoding` enum value that indicates this is a deleted file?

Reply:
> what about a ContentEncoding enum value that indicates this is a deleted file

I dig the compactness, but it seems _too_ clever. 👎